runtime-extension-model/src/test/java/org/mule/runtime/config/internal/validation/OperationDoesNotHaveDeprecatedExpressionTestCase.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.validation;
+
+import static org.mule.runtime.api.el.validation.ScopePhaseValidationItemKind.DEPRECATED;
+import static org.mule.test.allure.AllureConstants.MuleDsl.DslValidationStory.DSL_VALIDATION_STORY;
+import static org.mule.test.allure.AllureConstants.MuleDsl.MULE_DSL;
+import static org.mule.test.allure.AllureConstants.ReuseFeature.REUSE;
+import static org.mule.test.allure.AllureConstants.ReuseFeature.ReuseStory.OPERATIONS;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Optional.empty;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.AdditionalMatchers.not;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.mule.runtime.api.el.ExpressionLanguage;
+import org.mule.runtime.api.el.validation.Location;
+import org.mule.runtime.api.el.validation.ScopePhaseValidationItem;
+import org.mule.runtime.api.el.validation.ScopePhaseValidationItemKind;
+import org.mule.runtime.api.el.validation.ScopePhaseValidationMessages;
+import org.mule.runtime.ast.api.validation.Validation;
+import org.mule.runtime.ast.api.validation.ValidationResultItem;
+import io.qameta.allure.Features;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Stories;
+import io.qameta.allure.Story;
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+
+@Features({@Feature(MULE_DSL), @Feature(REUSE)})
+@Stories({@Story(DSL_VALIDATION_STORY), @Story(OPERATIONS)})
+public class OperationDoesNotHaveDeprecatedExpressionTestCase extends AbstractCoreValidationTestCase {
```

Comment:
add a test case where a deprecated expression is not used within a mule sdk operation as well